### PR TITLE
beancount: migrate to python@3.10

### PR DESCRIPTION
**Pull request by branchv**
beancount: migrate to python@3.10

### CODE REVIEW

**Comment by branchvincent** [comment]
`error in beautifulsoup4 setup command: use_2to3 is invalid.`

caused by https://setuptools.pypa.io/en/latest/history.html#v58-0-0

**Comment by Moisan** [comment]
Let's use their github related homepage for now since furius.ca is down.

File: Formula/beancount.rb [hunk 1]
```diff
@@ -167,8 +168,8 @@ class Beancount < Formula
   end
 
   resource "six" do
-    url "https://files.pythonhosted.org/packages/6b/34/415834bfdafca3c5f451532e8a8d9ba89a21c9743a0c59fbd0205c7f9426/six-1.15.0.tar.gz"
-    sha256 "30639c035cdb23534cd4aa2dd52c3bf48f06e5f4a941509c8bafd8ce11080259"
+    url "https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz"
+    sha256 "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926"
   end
```
Comment (carlocab) [inline Formula/beancount.rb:173]: `depends_on "six"`?

File: Formula/beancount.rb [hunk 2]
```diff
@@ -112,8 +118,8 @@ class Beancount < Formula
...
 
   resource "protobuf" do
-    url "https://files.pythonhosted.org/packages/c5/82/cee5dcde1c7a0ffe1336946a117d31b1a394558fcf4d8ca3fba720a47f80/protobuf-3.15.6.tar.gz"
-    sha256 "2b974519a2ae83aa1e31cff9018c70bbe0e303a46a598f982943c49ae1d4fcd3"
+    url "https://files.pythonhosted.org/packages/ff/3e/19b09fd98ca30a9cd53269806602ffe1d32464fe1f662ec8765a215f0495/protobuf-3.18.1.tar.gz"
+    sha256 "1c9bb40503751087300dd12ce2e90899d68628977905c76effc48e66d089391e"
   end
```
Comment (carlocab) [inline Formula/beancount.rb:123]: `depends_on "protobuf"`?

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**